rac2: misc token adjustment logging improvements

kvoli opened the pull request:
See individual commit messages. Changes were made in aide of debugging failures.

Part of: https://github.com/cockroachdb/cockroach/issues/128040
Release note: None

kvoli committed:
rac2: move adjust token vlog to adjust locked

When the `TokenCounter` tries to deduct tokens, via `TryDeduct`, it
retains a lock for the duration of the function, meaning that it must
call into `adjustLocked` directly, rather than indirectly via `adjust`.

The `v(2)` log statement which prints token adjustments was in `adjust`,
so missed this other source of adjustments, which are important for
debugging.

Move it over.

Part of: #128040
Release note: None

kvoli committed:
rac2: include normal flag in token adjustment logging

Otherwise, the log line reads `flag=`, because we omit printing the flag
when formatting the adjustment.

Part of: #128040
Release note: None

kvoli committed:
rac2: include token type in adjust logging

Previously, when adjustments were logged, they would not include whether
the adjustment was made for send or for eval tokens, which made it hard
to impossible to differentiate the log lines.

Include the token type in the log message.

Part of: #128040
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/132317)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 1 of 1 files at r1, 3 of 3 files at r2, 6 of 6 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/132317#-:-O8r4fK57bhyGBhZ5oy6:bg7igkv)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @kvoli)

___
*[`pkg/kv/kvserver/kvflowcontrol/rac2/token_counter.go` line 595 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/132317#-O8r3kTg9iff7kzev3uW:-O8r3kTh46KjMvG8Scbv:b-t72jmb) ([raw file](https://github.com/cockroachdb/cockroach/blob/7538619804e1b22dc3efdb36424102734df25e52/pkg/kv/kvserver/kvflowcontrol/rac2/token_counter.go#L595)):*
> ```Go
> 		t.metrics.onUnaccounted(unaccounted)
> 	}
> 	if log.V(2) {
> ```

`log.V(2)` is not super-cheap (I've seen it in profiles) and this is a shared `tokenCounter` mutex across many ranges. How about lifting this call into the callers before they acquire the mutex, and passing a `doLog` bool parameter.


<!-- Sent from Reviewable.io -->

kvoli committed:
rac2: check token adjustment v(2) outside lock

`V(2)` can be expensive on hot paths, this can be further compounded to
if done under a lock.

Determine if expensive logging (v=2) is enabled outside a lock and pass
this information into `adjustTokensLocked`.

Part of: #128040
Release note: None

kvoli reviewed (commented):
TYFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/132317#-:-O8rPU8sANVCFgA0LGMr:bzftgo7)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @sumeerbhola)

___
*[`pkg/kv/kvserver/kvflowcontrol/rac2/token_counter.go` line 595 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/132317#-O8r3kTg9iff7kzev3uW:-O8rPQrJ9c80D7MKI20g:b-s9px4m) ([raw file](https://github.com/cockroachdb/cockroach/blob/7538619804e1b22dc3efdb36424102734df25e52/pkg/kv/kvserver/kvflowcontrol/rac2/token_counter.go#L595)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

`log.V(2)` is not super-cheap (I've seen it in profiles) and this is a shared `tokenCounter` mutex across many ranges. How about lifting this call into the callers before they acquire the mutex, and passing a `doLog` bool parameter.
</blockquote></details>

Done -- pushed another commit with this change.


<!-- Sent from Reviewable.io -->

kvoli commented:
TYFTR

bors r=sumeerbhola
